app/queries/servers/channel.test.ts

```diff
@@ -0,0 +1,1552 @@
+
+import {General, Permissions} from '@constants';
+import {MM_TABLES} from '@constants/database';
+import ServerDataOperator from '@database/operator/server_data_operator';
+import {hasPermission} from '@utils/role';
+
+import {prepareChannels,
+    prepareMissingChannelsForAllTeams,
+    prepareMyChannelsForTeam,
+    prepareDeleteChannel,
+    prepareDeleteBookmarks,
+    type ChannelMembershipsExtended,
+    queryAllChannels,
+    queryAllChannelsForTeam,
+    queryAllChannelsInfo,
+    queryAllChannelsInfoForTeam,
+    queryAllMyChannel,
+    queryAllMyChannelsForTeam,
+    queryAllUnreadDMsAndGMsIds,
+    queryChannelsById,
+    queryChannelsByTypes,
+    queryUserChannelsByTypes,
+    queryTeamDefaultChannel,
+    queryMyChannelsByTeam,
+    queryUsersOnChannel,
+    getMyChannel,
+    observeMyChannel,
+    observeMyChannelRoles,
+    getChannelById,
+    observeChannel,
+    getChannelByName,
+    getDefaultChannelForTeam,
+    getCurrentChannel,
+    getCurrentChannelInfo,
+    observeCurrentChannel,
+    getChannelInfo,
+    deleteChannelMembership,
+    addChannelMembership,
+    getMembersCountByChannelsId,
+    queryAllMyChannelSettings,
+    queryMyChannelSettingsByIds,
+    queryMyChannelUnreads,
+    queryMyRecentChannels,
+    observeChannelInfo,
+    observeAllMyChannelNotifyProps,
+    observeNotifyPropsByChannels,
+    observeMyChannelMentionCount,
+    observeArchiveChannelsByTerm,
+    observeChannelsByLastPostAt,
+    observeChannelSettings,
+    observeDirectChannelsByTerm,
+    observeIsMutedSetting,
+    observeJoinedChannelsByTerm,
+    observeNotDirectChannelsByTerm,
+    queryChannelMembers,
+    queryChannelsForAutocomplete,
+    observeChannelMembers,
+} from './channel';
+import {queryRoles} from './role';
+import {getCurrentChannelId, observeCurrentChannelId, observeCurrentUserId} from './system';
+import {observeTeammateNameDisplay} from './user';
+
+import type {UserModel} from '@database/models/server';
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type ChannelBookmarkModel from '@typings/database/models/servers/channel_bookmark';
+import type ChannelInfoModel from '@typings/database/models/servers/channel_info';
+import type ChannelMembershipModel from '@typings/database/models/servers/channel_membership';
+import type MyChannelModel from '@typings/database/models/servers/my_channel';
+import type MyChannelSettingsModel from '@typings/database/models/servers/my_channel_settings';
+
+jest.mock('./role');
+jest.mock('./system');
+jest.mock('./user');
+jest.mock('@utils/role');
+
+describe('prepareChannels', () => {
+    let operator: ServerDataOperator;
+
+    beforeEach(() => {
+        operator = {
+            handleChannel: jest.fn(),
+            handleChannelInfo: jest.fn(),
+            handleChannelMembership: jest.fn(),
+            handleMyChannel: jest.fn(),
+            handleMyChannelSettings: jest.fn(),
+        } as unknown as ServerDataOperator;
```

Comment:
No need to mock the operator, the database is already mocked and all should work just fine